### PR TITLE
net/ieee802154: Make RAW mode generic

### LINKED ISSUE #5004: Normalize IEEE802514 driver "raw" mode.
All the other drivers implement a RAW mode, enabled by a config like IEEE802154_MCR20A_RAW.  We should do this on KW41Z as well.  Eventually at least the config option should get refactored up a level.

### PR DESCRIPTION
**Pull request by tbursztyka**
- Renaming NET_L2_RAW_CHANNEL to NET_RAW_MODE
- Create a generic IEEE 802.15.4 raw mode for drivers
- Modify the IEEE 802.15.4 drivers so it passes the packet unmodified,
up to code using that mode to apply the necessary changes on the
received net_pkt according to their needs
- Modify wpanusb/wpan_serial relevantly

Fixes #5004

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
Typo in commit message: s/Creat/Create/

File: subsys/net/ip/Kconfig [hunk 1]
```diff
@@ -10,11 +10,11 @@ menu "IP stack"
 	depends on NETWORKING
 
 # Hidden option
-config NET_L2_RAW_CHANNEL
+config NET_RAW_MODE
```
Comment (jukkar) [inline subsys/net/ip/Kconfig]: This config option has now become totally obscure as now even the name does not give any hint what it means. As there is no help option one cannot easily figure out what this means. I feel the name of the option is a bit too generic, but I do not know how we should call this.
Biggest issue is the missing help text or proper comment in Kconfig file, we really need to explain what this option is doing.
The rest of the PR looks fine.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: subsys/net/ip/Kconfig (hunk 1 above)
Comment (tbursztyka) [reply subsys/net/ip/Kconfig]: true, some doc would be indeed necessary here

Comment (finikorg) [reply subsys/net/ip/Kconfig]: +1 to @jukkar , maybe L2_RAW otherwise NET_RAW seems too vague.

Comment (tbursztyka) [reply subsys/net/ip/Kconfig]: No L2_RAW is semantically wrong. It's the full NET stack which basically is avoided. Only the very basic primitives are built-in (net_pkt so far...) So it's really about RAW networking: net drivers get to rx/tx net_pkt, but the whole logic to process what's inside net_pkt is not here.

**Comment by jukkar** [comment]
> RAW networking: net drivers get to rx/tx net_pkt, but the whole logic to process what's inside net_pkt is not here

The RAW word is ambiguous here, it is not clear what it means. What about calling this NET_BYPASS or similar?

**Comment by tbursztyka** [comment]
bypass would mean you could have both at runtime, with a possibility for driver to "bypass" the net stack. 

I mean, it's a hidden option, only meant for wpan right now, just making it saner. it used to have "raw" in the name before and that did not seem to be an issue. raw is kind of logical, it's like in socket when you do raw networking.

**Comment by jukkar** [comment]
I know, naming this thing is tricky. Anyway, the documentation of the option is more important than the actual name.

**Comment by tbursztyka** [comment]
yep, not sure I nailed the doc actually. Any propositions would be welcome

File: subsys/net/ip/Kconfig [hunk 2]
```diff
@@ -10,11 +10,16 @@ menu "IP stack"
...
 	bool
 	default n
+	help
+	This is a very specific option used to built only the very minimal
```
Comment (jukkar) [inline subsys/net/ip/Kconfig]: Should we describe here that
* user should not set this normally
* what applications (wpan_serial etc) use this and what purpose these applications set this value

Comment (tbursztyka) [reply subsys/net/ip/Kconfig]: I documented it a bit better, but the more important one was in ieee802154 where I reference wpan* stuff.

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by jhedberg** [comment]
This seems to have introduced the following Kconfig warning:

```
drivers/ieee802154/Kconfig.nrf5:9:warning: config symbol defined without type
```

Shouldn't CI catch this kind of stuff and prevent merging?

**Pull request merged**